packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -106,8 +226,13 @@ class TimeInput extends React.Component {
 
     // Prepend a 0 to the value when losing focus and the value is single digit.
     if (stateValue.length === 1) {
-      stateValue = '0'.concat(stateValue);
-      this.handleValueChange(event, type, stateValue);
+      if (this.props.variant === FORMAT_12_HOUR && type === inputType.HOUR && stateValue === '0') {
```

Comment:
Should there be another check for when input is '00' for 12-hour clock?
```
if (
   stateValue.length === 2
   && this.props.variant === FORMAT_12_HOUR
   && type === inputType.HOUR
   && stateValue === '00' 
) {
    stateValue = '12';
}
```

Reply:
I see you have it in lines 279-282 in handleHourChange(). I wonder if it would be cleaner to keep this logic handling in the same place?

Reply:
This logic actually isn't dealing with input change. This is the blur function, for when a user has inputted just one digit, and then left the input by clicking off or hitting the tab key. Thus we want to make that we only update the value if the value inside this function is only one character long.